orb-ui/src/main.rs

```diff
@@ -49,23 +49,45 @@ enum SubCommand {
     Daemon,
 
     /// Signup simulation
-    #[clap(action)]
-    Simulation,
+    #[clap(subcommand)]
+    Simulation(SimulationArgs),
 
     /// Recovery UI
     #[clap(action)]
     Recovery,
 }
+
+#[derive(Parser, Debug)]
+enum SimulationArgs {
+    /// Self-serve signup, app-based
+    #[clap(action)]
+    SelfServe,
+
+    /// Operator-based signup, with QR codes
+    #[clap(action)]
+    Operator,
+
+    /// show-car, infinite loop of signup
+    #[clap(action)]
+    ShowCar,
+}
+
 static HW_VERSION_FILE: OnceLock<String> = OnceLock::new();
 
-fn get_hw_version() -> Result<String> {
+#[derive(Debug, PartialEq)]
+enum Hardware {
+    Diamond,
+    Pearl,
+}
+
+fn get_hw_version() -> Result<Hardware> {
     let hw_file = HW_VERSION_FILE.get_or_init(|| {
         env::var("HW_VERSION_FILE")
             .unwrap_or_else(|_| "/usr/persistent/hardware_version".to_string())
     });
-    debug!("Reading HW version from {}", hw_file.as_str());
+    debug!("Reading hardware version from {}", hw_file.as_str());
 
-    String::from_utf8(
+    let hw =String::from_utf8(
         fs::read(hw_file.as_str())
```

Comment:
blocking in async. Although this happens at the start of the program and will eventually unblock, its still a potential footgun/hard to debug thing that may bite us later, so it makes sense to fix it.